app/controllers/players_controller.rb

```diff
@@ -34,6 +34,17 @@ def update
 
   def destroy
     player = Player.find(params[:id])
+    history = PlayerHistory.from_player(player)
+
+    # Delete each match/game in this player's history
+    history.each do |match|
+      StatService.new({
+        match: match,
+        winner: match.winner,
+        loster: match.loser
+      }).update_stats_for_deleted_match
+      match.destroy
```

Comment:
Looks like we can add a `before_destroy` callback to the model. See http://api.rubyonrails.org/classes/ActiveRecord/Callbacks.html